File: src/main/java/com/pisphere/governance/VotingSystem.java
package com.pisphere.governance;

import java.util.HashMap;
import java.util.Map;

public class VotingSystem {
    private Map<String, Integer> votes;

    public VotingSystem() {
        votes = new HashMap<>();
    }

    public void castVote(String proposal) {
        votes.put(proposal, votes.getOrDefault(proposal, 0) + 1);
        System.out.println("Vote cast for proposal: " + proposal);
    }

    public int getVotes(String proposal) {
        return votes.getOrDefault(proposal, 0);
    }
}

File: src/main/python/governance/voting_system.py
class VotingSystem:
    def __init__(self):
        self.votes = {}

    def cast_vote(self, proposal):
        self.votes[proposal] = self.votes.get(proposal, 0) + 1
        print(f"Vote cast for proposal: {proposal}. Total votes: {self.votes[proposal]}")

    def get_votes(self, proposal):
        return self.votes.get(proposal, 0)
